User guide

Esteban141 opened the pull request.

Esteban141 committed:
feat : merge + début de guide utilisateur

Esteban141 committed:
feat : modification architecture + ajout bouton

Changement dans l'organisation des pages relatives au guide. Ajout d'un bouton dans la page d'accueil pour avoir accès au guide. Suppression de l'autre chemin qui permettait d'accéder au guide.

Esteban141 committed:
fix : modification de la grille du guide

Différents ajustements pour rendre la grille visuellement plus agréable.

Esteban141 committed:
feat : finalisation page d'accueil du guide

Correction de différents problèmes de placement dans la grille de menu.
Ajout de la section de FAQ avec différentres questions déroulables.
Ainsi que différentes animations pour une meilleur UX (à vérifier).

Esteban141 committed:
Merge branch 'main' into Feature/TSKVOTERIGHT-1205

Esteban141 committed:
feat : intégration du design responsive

Adaptation de l'écran d'accueil du guide aux différents supports.
Notamment aux écrans de téléphone et de tablette.

Esteban141 committed:
fix : retouche readme.md

Esteban141 committed:
feat : adding guide section

Creation of the description page for each section of the guide. Use the
same css file for all sections.

Esteban141 committed:
feat : update of the guide sections

Changes on the css file themes.scss and adding JS scripts on every section to create smooth transitions in the sidebar

Esteban141 committed:
Merge branch 'main' into user-guide

Esteban141 committed:
feat : adding responsive design to guide sections

Various corrections to the guide and theme css files. JS code modifications to adapt guide sections. HTML structure changes to adapt to responsive design.

Kylian2 committed:
fix: css margin, font-size & position sticky

Kylian2 committed:
fix: images to material icon

Kylian2 commented on admin/pages/proposals/[id]/index.vue:
```diff
@@ -8,6 +8,85 @@
 
     <h1>{{ proposal['PRO_title_VC'] }}</h1>
 
+<<<<<<< HEAD
```
Un merge ou un conflit n'a pas été résolu ici. Il faut probablement conserver la version du MAIN.

Kylian2 commented on frontend/pages/tutorial/themes/account.vue:
```diff
@@ -0,0 +1,153 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Pas besoin de l'importer

Kylian2 replied:
La façon dont tu as codé tes composants n'est pas conforme à la façon de faire avec la syntaxe et les éléments apportés par VueJS. 

Il faut utiliser les références autant que possible

Par exemple, **pour gérer les classes dynamiques** plutôt que faire : 
```js
sidebar.classList.add('active')
```

Cela donnerait :
```js
//dans le template
:class="{ active: isSidebarOpen }

//dans le script
const isSidebarOpen = ref(false)
```

**Pour la gestion des événements**, à la place de 

```js
button.addEventListener('click', openSidebar);
````
Il faut utiliser : 

```js
<button @click="openSidebar">Un bouton </button>
```
(Ça ne fonctionne pas que sur des boutons évidement).

Il faudra apporter ces modifications sur toutes les pages...

Il y aurait probablement un moyen de factoriser le code du sommaire dans un composant mais je vais t'épargner cette peine pour l'instant.

Kylian2 commented on frontend/pages/tutorial/themes/gestion.vue:
```diff
@@ -0,0 +1,212 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Idem

Kylian2 commented on frontend/pages/tutorial/themes/groups.vue:
```diff
@@ -0,0 +1,154 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Idem

Kylian2 commented on frontend/pages/tutorial/themes/proposals.vue:
```diff
@@ -0,0 +1,154 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Idem

Kylian2 commented on frontend/pages/tutorial/themes/roles.vue:
```diff
@@ -0,0 +1,160 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Idem

Kylian2 commented on frontend/pages/tutorial/themes/votes.vue:
```diff
@@ -0,0 +1,138 @@
+</template>
+
+<script setup>
+import { onMounted } from 'vue'
```
Idem

Kylian2 commented on frontend/pages/tutorial/themes/account.vue:
```diff
@@ -0,0 +1,153 @@
+        </div>
+    </main>
+
+    <footer id="contact" class="footer">
```
Plutôt que de remettre le footer comme ça sur toutes les pages, crée un composant Footer et ajouter ce dernier. Cela permettra d'apporter de réduire la taille du code et d'apporter les modifications faites au footer une seule fois (et pas sur X pages). 

Ajoute également dans le footer une lien vers le guide.

Kylian2 commented on frontend/pages/tutorial/themes/account.vue:
```diff
@@ -0,0 +1,153 @@
+                    <ul class="theme__interactions__sidebar__summary">
+                        <li>
+                            <a href="#part1" class="theme__interactions__sidebar__summary__shortens">
+                                1. Gérer les paramètres de notification
```
À la place de numéroter à la main les éléments de la liste, utilise une liste ordonnée (peut-être as-tu déjà essayé et cela n'a pas donnée le résultat attendu ?) : 

```html
<ol>
    <li> item 1 </li>
    <li> item 2 </li>
    <li> item 3 </li>
</ol>
```

Kylian2 reviewed (changes requested):
Dans l'ensemble c'est du bon boulot, quelque petites coquilles sur l'utilisation de VueJS et quelques refactorisation de code à faire. 

**Liste des remarques**
- [ ] Mauvaise utilisation de VueJS pour les classes dynamiques et les écouteurs d'événement
- [ ] Numérotation manuelle (au lieu de liste ordonnée)
- [ ] Imports inutiles
- [ ] Créer un composant Footer (et ajouter un lien vers le guide)
- [ ] Problème de merge avec la branche `main` (il faut garder le contenu du `main`)

J'ai également apporté quelques modifications : sur les marges, j'ai mis le boutons retour en position `sticky` (au même titre que le sommaire), j'ai utiliser les icônes material-icon à la place des images (pour les flèches, croix, etc...).

Esteban141 committed:
Merge branch 'main' into user-guide

Esteban141 committed:
Merge branch 'user-guide' of https://github.com/Kylian2/vote-right into user-guide

Esteban141 committed:
refactor : CSS changes + improved vue usage

Made changes throughout the user guide to improve the usability of the Vue framework. At the same time, I made some minor changes to the user interface.

Kylian2 committed:
refactor: retrait import ref
